pandas/tests/io/parser/common.py

```diff
@@ -238,6 +238,17 @@ def test_csv_mixed_type(self):
         out = self.read_csv(StringIO(data))
         tm.assert_frame_equal(out, expected)
 
+    def test_csv_index_col_and_nrows(self):
+        data = """A,B,C
+1,1,1,2
+2,2,3,4
+3,3,4,5
+"""
+        out = self.read_csv(StringIO(data), low_memory=True, index_col=0,
+                            nrows=0)
+        tm.assert_index_equal(out.columns, pd.Index(['A', 'B', 'C']))
```

Comment:
Could you instead specify an expected `DataFrame` and use `tm.assert_frame_equal` to the output?

```
result = self.read_csv(...)
expected = pd.DataFrame(...)
tm.assert_frame_equal(result, expected)

```